controllers/storagecluster/reconcile.go

```diff
@@ -403,6 +404,26 @@ func (r *StorageClusterReconciler) reconcilePhases(
 		return reconcile.Result{}, nil
 	}
 
+	storageClusterPeerList := &ocsv1.StorageClusterList{}
+	err := r.Client.List(r.ctx, storageClusterPeerList, client.InNamespace(instance.Namespace))
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to list storageClusterPeer: %w", err)
+	}
+	for i := range storageClusterPeerList.Items {
+		scp := &storageClusterPeerList.Items[i]
+		lenOwners := len(scp.OwnerReferences)
+		err := controllerutil.SetOwnerReference(instance, scp, r.Scheme)
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("failed to set owner reference on StorageClusterPeer %v: %w", scp.Name, err)
+		}
+		if lenOwners != len(scp.OwnerReferences) {
+			err = r.Client.Update(r.ctx, scp)
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to update StorageClusterPeer %v: %w", scp.Name, err)
```

Comment:
Your message is very generic and basically just say that an update has failed. That same context comes from the internal error returned from the client. It will be preferable if your wrapping error would offer additional context
```suggestion
				return reconcile.Result{}, fmt.Errorf("Failed to persist StorageCluster owner ref on StorageClusterPeer %v: %w", scp.Name, err)
```